aws-sdk-core/lib/aws-sdk-core/credential_provider_chain.rb

```diff
@@ -67,9 +68,28 @@ def shared_credentials(options)
       nil
     end
 
+    def assume_role_credentials(options)
+      if Aws.shared_config.config_enabled?
+        if options[:config]
+          assume_role_with_profile(options[:config].profile, options[:config].region)
+        else
+          assume_role_with_profile(nil)
```

Comment:
This method call will fail with an incorrect number of arguments error.  I would suggest something like:

```ruby
def assume_role_credentials(options)
  if Aws.shared_config.config_enabled?
    profile = nil
    region = nil
    if options[:config]
      profile = options[:config].profile
      region = options[:config].region
    end
    assume_role_with_profile(profile, region)
  else
    nil
  end
end
```